default/python/AI/FreeOrionAI.py

```diff
@@ -122,49 +140,56 @@ def startNewGame(aggression_input=fo.aggression.aggressive):  # pylint: disable=
         debug("    Renaming to %s..." % new_name)
         res = fo.issueRenameOrder(planet_id, new_name)
         debug("    Result: %d; Planet is now named %s" % (res, planet.name))
-    _pre_game_start(empire.empireID, aistate)
+    _pre_game_start(empire.empireID)
 
 
 @error_handler
 def resumeLoadedGame(saved_state_string):  # pylint: disable=invalid-name
     """Called by client to when resume a loaded game."""
+    debug("Resuming loaded game")
+
+    if saved_state_string == "NO_STATE_YET" and fo.currentTurn() == 1:
+        info("AI given uninitialized state-string to resume from on turn 1.")
+        info("Assuming post-universe-generation autosave before any orders were sent"
+             "and behave as if a new game was started.")
+        return startNewGame()
+
     if fo.getEmpire() is None:
         fatal("This client has no empire. Doing nothing to resume loaded game.")
         return
 
     if fo.getEmpire().eliminated:
         info("This empire has been eliminated. Ignoring resume loaded game.")
         return
-    debug("Resuming loaded game")
-    if not saved_state_string:
-        error(
-            "AI given empty state-string to resume from; this is expected if the AI is assigned to an empire "
-            "previously run by a human, but is otherwise an error. AI will be set to Aggressive."
-        )
-        aistate = create_new_aistate(fo.aggression.aggressive)
-        aistate.session_start_cleanup()
+
+    aistate = None
+    if saved_state_string == "NOT_SET_BY_CLIENT_TYPE":
+        info("AI assigned to empire previously run by human.")
+        chat_human("We have been assigned an empire previously run by a human player. We can manage this.")
+    elif saved_state_string == "":
+        error("AI given empty state-string to resume from. "
+              "AI can continue but behaviour may be different from the previous session.")
     else:
         try:
             # loading saved state
             aistate = load_aistate(saved_state_string)
         except Exception as e:
-            # assigning new state
-            aistate = create_new_aistate(fo.aggression.aggressive)
-            aistate.session_start_cleanup()
             error(
-                "Failed to load the AIstate from the savegame. The AI will"
-                " play with a fresh AIstate instance with aggression level set"
-                " to 'aggressive'. The behaviour of the AI may be different"
-                " than in the original session. The error raised was: %s" % e,
-                exc_info=True,
+                "Failed to load the AIstate from the savegame: %s"
+                " AI can continue but behaviour may be different from the previous session.", e,
+                exc_info=True
             )
-    _pre_game_start(fo.getEmpire().empireID, aistate)
+    if aistate is None:
+        info("Creating new ai state due to failed load.")
```

Comment:
Shouldn't this be an error, too?

Reply:
I guess it can be argued either way. There are cases where we fail to load and must create a new state without it being a "real" error, e.g. being assigned an empire previously controlled by a human player.

My intention was to have error logs above where we fail unexpectedly but just info here when we recovered.

Reply:
We implement robustness.  If something went wrong we still try to do our best.  This problem could happen because of some save game incompatibility.

Reply:
@Cjkjvfnby Not sure if you are arguing for ```error``` or ```info``` here.

Any string that isn't specially handled above is sent to the decoder. If invalid/incompatible, an Exception will be raised and logged accordingly. All of that happens before reaching here.